bidi-scanner/lib/detector.mjs

```diff
@@ -0,0 +1,35 @@
+/* Adapted from:code with original license:
```

Comment:
No risk of losing the code this time: I just copied it, and gave proper attribution.